cpp/dolfinx/geometry/BoundingBoxTree.h

```diff
@@ -70,9 +70,15 @@ class BoundingBoxTree
 
   /// Return bounding box coordinates for a given node in the tree
   /// @param[in] node The bounding box node index
-  /// @return The bounding box where [0] is the lower corner and [1] is
-  /// the upper corner
-  std::array<std::array<double, 3>, 2> get_bbox(std::size_t node) const;
+  /// @return A copy of bounding box (lower_corner, upper_corner). Shape (2,3).
+  /// Flattened row-major.
+  std::array<double, 6> copy_bbox(std::size_t node) const;
+
+  /// Return bounding box coordinates for a given node in the tree
+  /// @param[in] node The bounding box node index
+  /// @return The bounding box (lower_corner, upper_corner) as a subspan of the
+  /// bounding box coordinates. Shape (2,3). Flattened row-major.
+  std::span<const double, 6> get_bbox(std::size_t node) const;
```

Comment:
I think this should be removed, and `copy_bbox` renamed to `get_bbox` unless profiling shows a significant performance benefit. Returning a copy is better than a cont reference/pointer if there isn't a compelling performance reason.

Reply:
It does not seem like there is a benefit performance wise. These are the timings with `get_bbox`:
```bash
Build tree (mus): 8812 11677 9573
Coll tree (mus): 553568 576671  561480
Build tree (mus): 8945 11141 9416
Coll tree (mus): 554286 572038  562611
Build tree (mus): 9071 12135 9514
Coll tree (mus): 555176 565760  558456
Build tree (mus): 9360 13424 10399
Coll tree (mus): 567018 748228  648080
```
(same code as in https://github.com/FEniCS/dolfinx/pull/2313#issuecomment-1239332493)